### ISSUE TITLE
Not consistend deteciton of invalid-sequence-index

### ISSUE THREAD

**Issue by JulienPalard** [PyCQA/pylint#1480]
Using pylint version 1.7.1 with the following file:

```
#!/usr/bin/env python3
"""OKest docstring.
"""


def not_foo():
    """OKest docstring.
    """
    random_dict = not_bar()

    # invalid-sequence-index:
    print([value["a"] for key, value in random_dict.items()])

    # No error:
    print([value["a"] for value in random_dict.values()])

    for key, value in random_dict.items():
        # invalid-sequence-index:
        print(key, value["a"])

    # No error:
    for value in random_dict.values():
        print(value["a"])

    # No error:
    for key in random_dict:
        print(random_dict[key]["a"])


def not_bar():
    """OKest docstring.
    """
    return {"Oh look": "A string"}
```
I'm getting:

```
E: 14, 0: Sequence index is not an int, slice, or instance with __index__ (invalid-sequence-index)
E: 21, 0: Sequence index is not an int, slice, or instance with __index__ (invalid-sequence-index)
```

As the code obviouslly touches `["a"]` every time I expected the error to be reported consistently in each cases.

**Comment by rogalski** [comment]
Thanks for submitting an issue!

Which Python version are you running?

**Comment by JulienPalard** [comment]
Python 3.5.3

**Comment by rafaeelaudibert** [comment]
Old issue, but this still happens, with the same code as stated before, with the following output in `pylint --version`:

```
pylint 2.2.2
astroid 2.1.0
Python 3.6.5 (default, May 11 2018, 13:30:17) 
[GCC 7.3.0]
```

In my code, I can reproduce it with:
```python
messageInfo = self.bot.sendMessage(
  group,
  msg,
  disable_notification=True,
  disable_web_page_preview=True,
  parse_mode='HTML')

identifiers.append((messageInfo['chat']['id'], messageInfo['message_id'])) # Error on this line
```
, whereas self.bot.SendMessage returns a dictionary (courtesy of [Python Telegram Bot](https://github.com/python-telegram-bot/python-telegram-bot))

**Comment by PCManticore** [comment]
Thanks for the additional example @rafaeelaudibert As a rule of thumb, if the issue is not closed, it's most likely not fixed (unless advertently)

**Comment by anmol27katyani** [comment]
What about now?

**Comment by JulienPalard** [comment]
@anmol27katyani what do you mean?

**Comment by clavedeluna** [comment]
The issue here is that for dict method `.values()` and for iterating over dict keys, [this line](https://github.com/PyCQA/pylint/blob/main/pylint/checkers/typecheck.py#L1664) is evaluated to `Uninferable`. Seems like this needs to be fixed in astroid.
